libraries/lib-utility/Tuple.h

```diff
@@ -0,0 +1,140 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  @file Tuple.h
+  @brief Extraction of sub-tuples of std::tuple (or other tuple-like classes)
+
+  Paul Licameli
+
+**********************************************************************/
+#ifndef __AUDACITY_TUPLE__
+#define __AUDACITY_TUPLE__
+
+#include <cstddef>
+#include <tuple>
+#include <utility>
+
+namespace Tuple {
+//! Type test metapredicate, for specializations of std::tuple only
+template<typename T, typename = void> struct is_tuple : std::false_type{};
+template<typename... Types> struct is_tuple<std::tuple<Types...>>
+   : std::true_type{};
+template<typename T> static constexpr auto is_tuple_v = is_tuple<T>::value;
+
+template<typename Tuple> constexpr bool Empty_v =
+   (0 == size_t(std::tuple_size_v<Tuple>));
+
+template<typename Tuple> [[nodiscard]] constexpr bool Empty(const Tuple &tuple)
```

Comment:
It's conventional in the C++ standard library to add that attribute to members named `empty`.  I think it is because of the English ambiguity between the adjective and the imperative verb.  One might mistakenly call `empty` when one means to call `clear`.

Reply:
Sounds plausible.